src/ray/object_manager/ownership_based_object_directory.cc

```diff
@@ -57,7 +57,8 @@ bool UpdateObjectLocations(const rpc::WorkerObjectLocationsPubMessage &location_
                            size_t *object_size) {
   bool is_updated = false;
   std::unordered_set<NodeID> new_node_ids;
-  // The size can be 0 if the update was a deletion. This assumes that an
+  // The size can be 0 if the update was a deletion. The size can also be unset if the
```

Comment:
If we decide to use -1 to indicate object size is unknown, we should do it globally meaning that we should change proto from uint64_t to int64_t as well?

Or should we change reference count to use 0 to indicate object size is unknown assuming object size should always greater than 0.

Reply:
I tried to make a patch to change everything to unsigned and use 0 for object size. But this line says 

    // The size can be 0 if the update was a deletion. This assumes that an
    // object's size is always greater than 0.

I'm not sure what this means. If it means the size being 0 is possible and is semantically different than "size unknown" we are out of luck. Any hints?

@stephanie-wang